lib/ret_web/channels/hub_channel.ex

```diff
@@ -372,14 +379,18 @@ defmodule RetWeb.HubChannel do
     if account |> can?(update_hub(hub)) do
       name_changed = hub.name != payload["name"]
       member_permissions_changed = hub.member_permissions != payload |> Hub.member_permissions_from_attrs()
+      can_change_privacy = account |> can?(update_hub_privacy(hub))
+      privacy_changed = can_change_privacy and hub.privacy != payload["privacy"]
 
       stale_fields = []
       stale_fields = if name_changed, do: ["name" | stale_fields], else: stale_fields
       stale_fields = if member_permissions_changed, do: ["member_permissions" | stale_fields], else: stale_fields
+      stale_fields = if privacy_changed, do: ["privacy" | stale_fields], else: stale_fields
 
       hub
       |> Hub.add_name_to_changeset(payload)
       |> Hub.add_member_permissions_to_changeset(payload)
+      |> add_privacy_to_changeset_if_allowed(can_change_privacy, payload)
```

Comment:
i think the idiom we use here is `try_add_privacy_to_changeset` -- also I would push the permission check into that function (and pass along account, hub) instead of passing in the result of the check